aws/resource_aws_storagegateway_working_storage.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/storagegateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsStorageGatewayWorkingStorage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsStorageGatewayWorkingStorageCreate,
+		Read:   resourceAwsStorageGatewayWorkingStorageRead,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"disk_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"gateway_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+		},
+	}
+}
+
+func resourceAwsStorageGatewayWorkingStorageCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).storagegatewayconn
+
+	diskID := d.Get("disk_id").(string)
+	gatewayARN := d.Get("gateway_arn").(string)
+
+	input := &storagegateway.AddWorkingStorageInput{
+		DiskIds:    []*string{aws.String(diskID)},
+		GatewayARN: aws.String(gatewayARN),
+	}
+
+	log.Printf("[DEBUG] Adding Storage Gateway working storage: %s", input)
+	_, err := conn.AddWorkingStorage(input)
+	if err != nil {
+		return fmt.Errorf("error adding Storage Gateway working storage: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", gatewayARN, diskID))
+
+	return resourceAwsStorageGatewayWorkingStorageRead(d, meta)
+}
+
+func resourceAwsStorageGatewayWorkingStorageRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).storagegatewayconn
+
+	gatewayARN, diskID, err := decodeStorageGatewayWorkingStorageID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &storagegateway.DescribeWorkingStorageInput{
+		GatewayARN: aws.String(gatewayARN),
+	}
+
+	log.Printf("[DEBUG] Reading Storage Gateway working storage: %s", input)
+	output, err := conn.DescribeWorkingStorage(input)
+	if err != nil {
+		if isAWSErrStorageGatewayGatewayNotFound(err) {
+			log.Printf("[WARN] Storage Gateway working storage %q not found - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading Storage Gateway working storage: %s", err)
+	}
+
+	if output == nil || len(output.DiskIds) == 0 {
+		log.Printf("[WARN] Storage Gateway working storage %q not found - removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	found := false
```

Comment:
nitpick `var found bool` would begin false... I'm okay either way :)

Reply:
Go newcomers might be confused by [variable type zero-value behavior](https://tour.golang.org/basics/12) -- explicit > implicit IMO since it doesn't cost any extra lines of code. 👍